pkg/sqlcache/informer/factory/informer_factory.go

```diff
@@ -129,24 +129,52 @@ func (f *CacheFactory) CacheFor(ctx context.Context, fields [][]string, external
 	}
 	f.informersMutex.Unlock()
 
+	start, err := f.getOrCreateCache(ctx, gi, fields, externalUpdateInfo, selfUpdateInfo, transform, client, gvk, namespaced, watchable)
+	defer func() {
+		if !start.IsZero() {
+			log.Infof("CacheFor IS DONE creating informer for %v (took %v)", gvk, time.Since(start))
+		}
+	}()
+	if err != nil {
+		f.mutex.RUnlock()
+		return nil, err
+	}
+
+	if !cache.WaitForCacheSync(f.ctx.Done(), gi.informer.HasSynced) {
+		return nil, fmt.Errorf("failed to sync SQLite Informer cache for GVK %v", gvk)
+	}
```

Comment:
WaitForCacheSync has been moved outside to facilitate the migration to per-GVK in the next PR. Essentially, we're going to be avoiding a deadlock by not having it behind the gi.mutex lock.

Reply:
Pulled this out as it's only needed for the next PR.